runtime/parachains/src/disputes.rs

```diff
@@ -911,26 +908,8 @@ impl StatementSetFilter {
 
 impl<T: Config> Pallet<T> {
 	/// Called by the initializer to initialize the disputes module.
-	pub(crate) fn initializer_initialize(now: T::BlockNumber) -> Weight {
-		let config = <configuration::Pallet<T>>::config();
-
-		let mut weight = Weight::zero();
-		for (session_index, candidate_hash, mut dispute) in <Disputes<T>>::iter() {
-			weight += T::DbWeight::get().reads_writes(1, 0);
-
-			if dispute.concluded_at.is_none() &&
-				dispute.start + config.dispute_conclusion_by_time_out_period < now
-			{
-				Self::deposit_event(Event::DisputeTimedOut(candidate_hash));
-
-				dispute.concluded_at = Some(now);
```

Comment:
> As discussed with @eskimor we don't really need the timeout event

Could you post the gist of the discussion here? This code not only removes timeout *events*, but the notion of timing out for disputes. I'm lacking the context here. cc @burdges

Reply:
Yes:

1. Disputes that hit the runtime should actually never time out - they have been confirmed, so they should conclude.
2. We don't have a timeout on the node side. So timing them out in the runtime is kind of pointless and out of sync with what the node would assume about the dispute.
3. We drop disputes after 6 sessions anyway.
4. We can not and should not slash anybody on a timed out dispute - the people who voted are the good guys (most likely).

Reply:
> We drop disputes after 6 sessions anyway.

Yes, but maybe we should be reverting a block if the dispute hasn't concluded instead of finalizing it after some time?

Reply:
Yes, what we could and probably should do: Revert [immediately once f+1 validators voted against a candidate](https://github.com/paritytech/polkadot/issues/6950).